pkg/challenges/challenges.go

```diff
@@ -69,6 +70,78 @@ type ChallengeResult struct {
 	AdditionalInfo map[AdditionalInfo]string
 }
 
+func (cr *ChallengeResult) Name(context.Context) string {
+	return cr.Value
```

Comment:
Proof of possession is checked in `challenges.ExtractSubject`, correct?

Reply:
`preChallengeResult.Subject` is meant to be an internal struct just used for ExtractSubject, so `CheckSignature(publicKey, challenge, result.Subject)` where `result` is a `preChallengeResult` seems correct to me.

Reply:
The end state should look something like this I think:

```golang
// Inside of our api.CreateSigningCertificate handler

// ip is an identity.IssuerPool and we authenticate and extract the principal first
principal, err := ip.Authenticate(ctx, rawIDToken)
if err != nil {
   // handle err
}

if csr != nil {
  err = csr.CheckSignature()
  if err != nil {
    // handle errpr
  }
} else {
  // Proof of possession path
  err = challenges.CheckSignature(publicKey, proofOfPosession, principal.Name(ctx))
  if err != nil {
    // handle err
  }
}
```

So basically: `ExtractSubject` will go away and we'll be using `CheckSignature` directly. Once this current change is in I can actually do that part so that there is a clear difference between parsing id token -> Principal and checking proof of possession.

Reply:
Sorry, might have pre-maturely resolved this convo. Feels related to the other thread below as well. Is the intent to use `Name()` to deal the proof of possession checking for a principal clear above?

Reply:
Yep, makes sense, this question just arose from this being an intermediate state. Sounds good to push that logic up to the handler.